### ISSUE TITLE
Service display by service_severity doesn't take care of dependencies

### ISSUE THREAD

**Issue by yasa1987** [Icinga/icinga2#10340]
Hello, I noticed that Dependencies are not taken in account when services are displayed in the dashboard, even if the "sort=service_severity" is the in the URL.

Services are also not displayed as "handled". 

I'm monitoring a network router chassis that has line cards (iom) in which we plug modules (mda) that contain the physical ports. The dependencies between services are configured as follows :
- Line card (iom) depends on the host
- Module (mda) depends on the line card (iom) in which it is plugged in
- Port depends on the module (mda) 

In short terms : port depends on mda who depends on iom

In my scenario, I plugged out the line card (iom) to simulate a failure. The attached modules (mda) and ports will check script will return a CRITICAL state as they only can work if the line card is present.

## Expected Behavior
When the line card (iom) service is CRITICAL, the attached modules (mda) services should be HANDLED if they also are in CRITICAL state, and the ports services should also be HANDLED if they are critical.

The line card (iom) service should be at the top of the list as other services depends on it.

## Current Behavior
The module (mda) services who depend on the CRITICAL line card are not marked as handled.
The port services who depend on the CRITICAL modules (mda) services are not marked as handled.

The root cause of the issue (CRITICAL iom service) is not at the top of the service problem list.

The strange thing is that the `icingacli monitoring list services` command puts the "iom" CRITICAL service at the top of the list. 
 
## Context
My router chassis (Nokia 7750 SR-7) is composed of line cards (iom) that contains modules (mda) who contains ports. All those elements are monitored with their own service. 
If a module (mda) fails, all the ports will be in CRITICAL state, but it's because of the mda failure. The MDA failure must be on top of issues list.
Same thing for the line card (iom) regarding MDA.

Service-to-Service dependency should do the job (I suppose their are meant for that, right?)

## Your Environment
* Icinga Web 2 version and modules (System - About): 2.5.1
* Version used (`icinga2 --version`): r2.8.1-1
* Operating System and version: Debian 3.16.51-3+deb8u1 (2018-01-08) x86_64 GNU/Linux
* Enabled features (`icinga2 feature list`): api checker command ido-mysql influxdb mainlog notification statusdata
* Config validation (`icinga2 daemon -C`):
information/cli: Icinga application loader (version: r2.8.1-1)
information/cli: Loading configuration file(s).
information/ConfigItem: Committing config item(s).
information/ApiListener: My API identity: labradoor.tvssa.net
warning/ApplyRule: Apply rule 'mail-ingtvs-host' (in /etc/icinga2/tvs.d/notifications.conf: 16:1-16:45) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'mail-ingtvs-service' (in /etc/icinga2/tvs.d/notifications.conf: 22:1-22:51) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'sms-global-host' (in /etc/icinga2/tvs.d/notifications.conf: 30:1-30:44) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'sms-global-service' (in /etc/icinga2/tvs.d/notifications.conf: 36:1-36:50) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'mail-giti-host' (in /etc/icinga2/tvs.d/notifications.conf: 44:1-44:43) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'mail-giti-service' (in /etc/icinga2/tvs.d/notifications.conf: 50:1-50:49) for type 'Notification' does not match anywhere!
warning/ApplyRule: Apply rule 'huawei_powersupply' (in /etc/icinga2/tvs.d/services_huawei_hardware.conf: 15:1-15:34) for type 'Service' does not match anywhere!
information/ConfigItem: Instantiated 1 ApiListener.
information/ConfigItem: Instantiated 3 Zones.
information/ConfigItem: Instantiated 1 Endpoint.
information/ConfigItem: Instantiated 1 FileLogger.
information/ConfigItem: Instantiated 254 Notifications.
information/ConfigItem: Instantiated 6 NotificationCommands.
information/ConfigItem: Instantiated 249 CheckCommands.
information/ConfigItem: Instantiated 13 HostGroups.
information/ConfigItem: Instantiated 1 IcingaApplication.
information/ConfigItem: Instantiated 10 Hosts.
information/ConfigItem: Instantiated 5 UserGroups.
information/ConfigItem: Instantiated 42 Dependencies.
information/ConfigItem: Instantiated 8 Users.
information/ConfigItem: Instantiated 3 TimePeriods.
information/ConfigItem: Instantiated 155 Services.
information/ConfigItem: Instantiated 5 ServiceGroups.
information/ConfigItem: Instantiated 1 CheckerComponent.
information/ConfigItem: Instantiated 1 StatusDataWriter.
information/ConfigItem: Instantiated 1 ExternalCommandListener.
information/ConfigItem: Instantiated 1 IdoMysqlConnection.
information/ConfigItem: Instantiated 2 InfluxdbWriters.
information/ConfigItem: Instantiated 1 NotificationComponent.
information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
information/cli: Finished validating the configuration file(s).

**Comment by julianbrost** [comment]
@nilmerg (or maybe @flourish86) as that issue previously was in the Icinga Web project and you even considered fixing it there, did you already think about how severe unreachability is, i.e. how it compares with other factors like downtimes or acknowledgements?

But if in doubt, it will probably be done similar to how it's done for the host severity already.

**Comment by nilmerg** [comment]
No. Icinga DB Web enhanced this with v1.1, though. The OP is talking about the monitoring module anyway, which was the only available thing at the time. In case Icinga DB includes reachability in the severity, Web doesn't need to do anything.
